core/src/test/java/org/elasticsearch/index/mapper/core/BooleanFieldTypeTests.java

```diff
@@ -32,4 +32,13 @@ protected MappedFieldType createDefaultFieldType() {
     public void setupProperties() {
         setDummyNullValue(true);
     }
+
+    public void testValueForSearch() {
+        MappedFieldType ft = createDefaultFieldType();
+        assertEquals(true, ft.valueForSearch("T"));
+        assertEquals(false, ft.valueForSearch("F"));
+        expectThrows(IllegalArgumentException.class, () -> ft.valueForSearch(0));
```

Comment:
These should probably check the message on the exception so we know we got the correct IAE?